backend/handlers/math/outcomes/dbpm/marketshares_test.go

```diff
@@ -1,551 +1,709 @@
+				if payout.Payout != expected.Payout || payout.Outcome != expected.Outcome {
+					t.Errorf(
+						"%s: payout %d mismatch. Expected {Payout: %.17g, Outcome: %s}, got {Payout: %.17g, Outcome: %s}",
+						tc.Name, i, expected.Payout, expected.Outcome, payout.Payout, payout.Outcome,
+					)
 				}
 			}
+		})
+	}
 
-			// Check for any unexpected extra users in the results
-			for username := range resultsMap {
-				if _, ok := expectedResults[username]; !ok {
-					t.Errorf("Test %s failed: unexpected position for username %s", tc.Name, username)
-				}
+}
+
+func TestCalculateNormalizationFactorsDBPM(t *testing.T) {
+	testcases := []struct {
+		Name          string
+		CoursePayouts []CourseBetPayout
+		S_YES         int64
+		S_NO          int64
+		ExpectedF_YES float64
+		ExpectedF_NO  float64
+	}{
+		{
+			Name:          "InitialMarketState",
+			CoursePayouts: nil,
+			S_YES:         0,
+			S_NO:          0,
+			ExpectedF_YES: 0,
+			ExpectedF_NO:  0,
+		},
+		{
+			Name: "FirstBetNoDirection",
+			CoursePayouts: generateCoursePayouts(
+				[]float64{0}, // Payout = |0.167 - 0.167| * 20
+				[]string{"NO"},
+			),
+			S_YES:         3,
+			S_NO:          17,
+			ExpectedF_YES: 0,
+			ExpectedF_NO:  0,
+		},
+		{
+			Name: "SecondBetYesDirection",
+			CoursePayouts: generateCoursePayouts(
+				[]float64{4.1600000000000001, 0},
+				[]string{"NO", "YES"},
+			),
+			S_YES:         11,
+			S_NO:          19,
+			ExpectedF_YES: 0,
+			ExpectedF_NO:  4.5673076923076925,
+		},
+		{
+			Name: "ThirdBetYesDirection",
+			CoursePayouts: generateCoursePayouts(
+				[]float64{6.6599999999999993, 1.25, 0},
+				[]string{"NO", "YES", "YES"},
+			),
+			S_YES:         20,
+			S_NO:          20,
+			ExpectedF_YES: 16,
+			ExpectedF_NO:  3.0030030030030033,
+		},
+		{
+			Name: "FourthBetNegativeNoDirection",
+			CoursePayouts: generateCoursePayouts(
+				[]float64{9.1600000000000001, 2.5, 1.25, 0},
+				[]string{"NO", "YES", "YES", "NO"},
+			),
+			S_YES:         19,
+			S_NO:          11,
+			ExpectedF_YES: 5.066666666666666,
+			ExpectedF_NO:  1.2008733624454149,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.Name, func(t *testing.T) {
+			actualF_YES, actualF_NO := CalculateNormalizationFactorsDBPM(tc.S_YES, tc.S_NO, tc.CoursePayouts)
+
+			// Log the results for manual verification
+			t.Logf("%s: F_YES=%f, F_NO=%f", tc.Name, actualF_YES, actualF_NO)
+
+			if actualF_YES != tc.ExpectedF_YES || actualF_NO != tc.ExpectedF_NO {
+				t.Errorf(
+					"%s: expected F_YES=%f, F_NO=%f; got F_YES=%f, F_NO=%f",
+					tc.Name, tc.ExpectedF_YES, tc.ExpectedF_NO, actualF_YES, actualF_NO,
+				)
 			}
 		})
 	}
+
 }
 
-func TestCalculateCoursePayoutsDBPM(t *testing.T) {
+func TestCalculateScaledPayoutsDBPM(t *testing.T) {
 	testcases := []struct {
-		Name               string
-		Bets               []models.Bet
-		ProbabilityChanges []wpam.ProbabilityChange
-		CoursePayouts      []CourseBetPayout
+		Name                  string
```

Comment:
we don't need to include `tc.Name` because the error message is output underneath TestFunc/TestCaseName